docs/book/shipments.rst

```diff
@@ -4,49 +4,139 @@
+therefore while configuring your **ShippingMethods** pay special attention to the zones you are assigning to them.
+You have to prepare methods for each zone, because the available methods are retrieved for the zone the customer has basing on his address.
 
-Examples
---------
+Shipping Cost Calculators
+-------------------------
 
-...
+The shipping cost calculators are services that are used to calculate the cost for a given shipment.
 
-Product and ProductVariant Configuration
-----------------------------------------
+The `CalculatorInterface <https://github.com/Sylius/Sylius/blob/master/src/Sylius/Component/Shipping/Calculator/CalculatorInterface.php>`_
+has a method ``calculate()`` that takes object with a configuration and returns *integer* that is the cost of shiping for that subject.
+It also has a ``getType()`` method that works just like in the forms.
 
-...
+To select a proper service we have a one that decides for us
+- the `DelegatingCalculator <https://github.com/Sylius/Sylius/blob/master/src/Sylius/Component/Shipping/Calculator/DelegatingCalculator.php>`_.
+Basing on the **ShippingMethod** assigned on the Shipment it will get its calculator type and configuration and calculate the cost properly.
 
-Final Thoughts
---------------
+.. code-block:: php
+
+    $this->container->get('sylius.repository.shipment')->add($shipment);
+
+    $shippingCalculator = $this->container->get('sylius.shipping_calculator');
+
+    $cost = $shippingCalculator->calculate($shipment);
+
+Built-in Calculators
+--------------------
+
+Shipping/Calculator
+The already defined calculators in Sylius are described as constants in the
+`Sylius\Component\Shipping\Calculator\DefaultCalculators <https://github.com/Sylius/Sylius/blob/master/src/Sylius/Component/Shipping/Calculator/DefaultCalculators.php>`_
+
+* **FlatRateCalculator** - just returns the ``amount`` from the ShippingMethod's configuration.
+* **FlexibleRateCalculator** - on the ShippingMethod's configuration it should have the ``first_unit_cost``, ``additional_unit_cost`` and ``additional_unit_limit`` configured - it is useful if you want to have different cost for teh fist item than for the rest.
+* **PerUnitRateCalculator** - returns the ``amount`` from the ShippingMethod's configuration multiplied by the ``units`` count.
+* **VolumeRateCalculator** - returns the ``amount`` from the ShippingMethod's configuration multiplied by the ``units`` count divided by the configured ``division``.
+* **WeightRateCalculator** - returns the ``fixed`` from the ShippingMethod's configuration multiplied plus the ``variable`` multiplied by the ``weight`` of the shipment and divided by the ``division`` from the configuration.
+
+
+Shipment Calculator Configuration
+---------------------------------
+
+As it was mentioned above the calculators may have different configuration in an array. Such keys are available:
```

Comment:
Different keys are available for different calculators, we should not list them here. We will have reference with all shipping calculators and their configurations later. Please remove this part.